src/admin/components/ResourceTabs/PolygonReviewTab/index.tsx

```diff
@@ -0,0 +1,620 @@
+import { uploadImageData } from "@/pages/site/[uuid]/components/MockecData";
+import { EntityName, FileType, UploadedFile } from "@/types/common";
+
+import SitePolygonReviewAside from "./components/PolygonReviewAside";
+import { IpolygonFromMap } from "./components/Polygons";
+import SitePolygonStatus from "./components/SitePolygonStatus/SitePolygonStatus";
+
+interface IProps extends Omit<TabProps, "label" | "children"> {
+  type: EntityName;
+  label: string;
+}
+export interface IPolygonItem {
+  id: string;
+  status: "draft" | "submitted" | "approved" | "needs-more-information";
+  label: string;
+  uuid: string;
+}
+
+interface TableItemMenuProps {
+  ellipse: boolean;
+  "planting-start-date": string | null;
+  "polygon-name": string;
+  "restoration-practice": string;
+  source?: string;
+  "target-land-use-system": string | null;
+  "tree-distribution": string | null;
+  uuid: string;
+}
+
+interface DeletePolygonProps {
+  uuid: string;
+  message: string;
+}
+
+const PolygonReviewAside: FC<{
+  type: EntityName;
+  data: IPolygonItem[];
+  polygonFromMap: IpolygonFromMap;
+  setPolygonFromMap: any;
+  refresh?: () => void;
+  mapFunctions: any;
+}> = ({ type, data, polygonFromMap, setPolygonFromMap, refresh, mapFunctions }) => {
+  switch (type) {
+    case "sites":
+      return (
+        <SitePolygonReviewAside
+          data={data}
+          polygonFromMap={polygonFromMap}
+          setPolygonFromMap={setPolygonFromMap}
+          mapFunctions={mapFunctions}
+          refresh={refresh}
+        />
+      );
+    default:
+      return null;
+  }
+};
+
+const PolygonReviewTab: FC<IProps> = props => {
+  const { isLoading: ctxLoading, record } = useShowContext();
+  const [files, setFiles] = useState<UploadedFile[]>([]);
+  const [saveFlags, setSaveFlags] = useState<boolean>(false);
+  const [isUserDrawing, setIsUserDrawing] = useState<boolean>(false);
+
+  const [polygonFromMap, setPolygonFromMap] = useState<IpolygonFromMap>({ isOpen: false, uuid: "" });
+
+  async function storePolygon(geojson: any, record: any) {
+    if (geojson?.length) {
+      const response = await fetchPostV2TerrafundPolygon({
+        body: { geometry: JSON.stringify(geojson[0].geometry) }
+      });
+      const polygonUUID = response.uuid;
+      if (polygonUUID) {
+        const site_id = record.uuid;
+        await fetchPostV2TerrafundSitePolygonUuidSiteUuid({
+          body: {},
+          pathParams: { uuid: polygonUUID, siteUuid: site_id }
+        }).then(() => {
+          refetch();
+          setPolygonFromMap({ uuid: polygonUUID, isOpen: true });
+        });
+      }
+    }
+  }
+
+  const mapFunctions = useMap(storePolygon);
+
+  const { data: sitePolygonData, refetch } = useGetV2SitesSitePolygon<SitePolygonsDataResponse>({
+    pathParams: {
+      site: record.uuid
+    }
+  });
+
+  const { data: sitePolygonBbox, refetch: refetchSiteBbox } = useGetV2SitesSiteBbox({
+    pathParams: {
+      site: record.uuid
+    }
+  });
```

Comment:
When I see two nearly identical GET requests in a component it automatically makes me wonder if the responses from these endpoints could be combined into a single request. Especially since our target audience is potentially on a bad internet connection, being efficient about these things is generally a win. Please look into if these two can be combined.